contributors/devel/sig-architecture/api-conventions.md

```diff
@@ -866,6 +867,17 @@ language client, and any other clients that use corresponding types
 Therefore, we ask that pointers always be used with optional fields that do not
 have a built-in `nil` value.
 
+## Nullable
+
+The `+nullable` comment tag allows the json `null` value to be a valid value
+for a field. The `null` value is serialized only when a field is a pointer
+in the Go definition, and does not have the `omitempty` json tag.
+
+For example, a map field marked with `+nullable` would accept either `foo: null` or `foo: {}`.
+
+Usage of `+nullable` is discouraged as it introduces several issues:
+- It is not compatible with json merge patching.
+- It is not compatible with generic protobuf.
 
```

Comment:
I _think_ it's also not compatible with server-side apply, but this should be verified (I have a vague memory of SSA using null values as a signal to remove a key, which would make a null value impossible to persist via SSA)

Reply:
I've reached out to folks in sig-apimachinery for a response for this one, hoping they will have seen this and can give me a quick answer

Reply:
Looking at an openshift type that has Nullable, the generated applyconfiguration looks like 
```
type ClusterResourceQuotaSelectorApplyConfiguration struct {
	LabelSelector      *metav1.LabelSelectorApplyConfiguration `json:"labels,omitempty"`
	AnnotationSelector map[string]string                       `json:"annotations,omitempty"`
}
```
If I were to explicitly set a `null`, Idon't think this could round trip through this type given the omitempty.
My understanding of the generated applyconfiguration types is that they are always pointer+omitempty (apart from maps/slices which are just omitempty), and `omitempty` is not compatible with `+nullable` right?